wcfsetup/install/files/lib/data/user/UserProfileAction.class.php

```diff
@@ -609,6 +614,83 @@ public function validateUploadCoverPhoto()
         $uploadHandler->validateFiles(new UserCoverPhotoUploadFileValidationStrategy());
     }
 
+    /**
+     * Sets an avatar for a given user.
+     *
+     * @throws UserInputException If none or more than one user is given.
+     * @throws \InvalidArgumentException If the given avatar is not an image or is incorrect sized.
+     * @throws \RuntimeException If the avatar could not be saved for any reasons.
+     */
+    public function setAvatar(): array
+    {
+        $user = $this->getSingleObject();
+
+        $imageData = \getimagesize($this->parameters['fileLocation']);
+
+        if (!$imageData) {
+            throw new \InvalidArgumentException("The given file is not an image.");
+        }
+
+        if ($imageData[0] != UserAvatar::AVATAR_SIZE || $imageData[1] != UserAvatar::AVATAR_SIZE) {
+            throw new \InvalidArgumentException(
+                "The given avatar has not the size of " . UserAvatar::AVATAR_SIZE . "×" . UserAvatar::AVATAR_SIZE . "."
```

Comment:
```suggestion
                "The given avatar does not have the size of " . UserAvatar::AVATAR_SIZE . "x" . UserAvatar::AVATAR_SIZE . "."
```

No unicode characters, please.